src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java

```diff
@@ -278,7 +278,8 @@ public void databaseClosing() {
     protected void close() {
         // shutdown ScheduledExecutorService. Without shutdown, cause thread leak
         if (remoteRequestExecutor != null && !remoteRequestExecutor.isShutdown()) {
-            Utils.shutdownAndAwaitTermination(remoteRequestExecutor);
+            // Note: Time to wait is set 60 sec because RemoteRequest's socket timeout is set 60 seconds.
+            Utils.shutdownAndAwaitTermination(remoteRequestExecutor, 60);
         }
```

Comment:
Switch to `openConnection()` and to support abort connection are tracked by different ticket. https://github.com/couchbase/couchbase-lite-java-core/issues/532